monkey/monkey_island/cc/resources/agent_configuration.py

```diff
@@ -0,0 +1,38 @@
+import json
+
+import marshmallow
+from flask import make_response, request
+
+from common.configuration.agent_configuration import AgentConfigurationSchema
+from monkey_island.cc.repository import IAgentConfigurationRepository
+from monkey_island.cc.resources.AbstractResource import AbstractResource
+from monkey_island.cc.resources.request_authentication import jwt_required
+
+
+class AgentConfiguration(AbstractResource):
+    urls = ["/api/agent-configuration"]
+
+    def __init__(self, agent_configuration_repository: IAgentConfigurationRepository):
+        self._agent_configuration_repository = agent_configuration_repository
+        self._schema = AgentConfigurationSchema()
+
+    @jwt_required
+    def get(self):
+        configuration = self._agent_configuration_repository.get_configuration()
+        configuration_json = self._schema.dumps(configuration)
+        return make_response(configuration_json, 200)
+
+    @jwt_required
+    def post(self):
+
+        try:
+            request_contents = json.loads(request.data)
+
+            configuration_object = self._schema.load(request_contents)
+            self._agent_configuration_repository.store_configuration(configuration_object)
+            return make_response({}, 200)
+        except (marshmallow.exceptions.ValidationError, json.JSONDecodeError) as err:
+            return make_response(
+                {"message": f"Invalid configuration supplied: {err}"},
+                400,
+            )
```

Comment:
Can we just use `self._schema.loads()`? If so, do we still need to check for `JSONDecodeError`?

```suggestion
            configuration_object = self._schema.loads(request.data)
            self._agent_configuration_repository.store_configuration(configuration_object)
            return make_response({}, 200)
        except marshmallow.exceptions.ValidationError as err:
            return make_response(
                {"message": f"Invalid configuration supplied: {err}"},
                400,
            )
```

Reply:
Without `json.JSONDecodeError` an invalid json returns 500 status. We need to handle  the JSONDecode error .
![image](https://user-images.githubusercontent.com/15820737/175316377-6b9ae391-1358-48b4-8b2f-536cab7a6f6a.png)

Reply:
We can still simplify the deserialization logic, though.